packages/create-skybridge/src/index.ts

```diff
@@ -0,0 +1,178 @@
+const helpMessage = `\
+Usage: create-skybridge [OPTION]... [DIRECTORY]
+
+Create a new Skybridge project by cloning the starter template.
+
+Options:
+  -h, --help              show this help message
+  --overwrite             remove existing files in target directory
+
+Examples:
+  create-skybridge my-app
+  create-skybridge . --overwrite
+`;
+
+function run([command, ...args]: string[], options?: SpawnOptions) {
+  if (!command) {
+    throw new Error("Command is required");
+  }
+  const { status, error } = spawnSync(command, args, options);
+  if (status != null && status > 0) {
+    process.exit(status);
+  }
+
+  if (error) {
+    console.error(`\n${command} ${args.join(" ")} error!`);
+    console.error(error);
+    process.exit(1);
+  }
+}
+
+async function init() {
+  const argTargetDir = argv._[0]
+    ? formatTargetDir(String(argv._[0]))
+    : undefined;
+  const argOverwrite = argv.overwrite;
+
+  const help = argv.help;
+  if (help) {
+    console.log(helpMessage);
+    return;
+  }
+
+  const interactive = process.stdin.isTTY;
+  const cancel = () => prompts.cancel("Operation cancelled");
+
+  // 1. Get project name and target dir
+  let targetDir = argTargetDir;
+  if (!targetDir) {
+    if (interactive) {
+      const projectName = await prompts.text({
+        message: "Project name:",
+        defaultValue: defaultProjectName,
+        placeholder: defaultProjectName,
+        validate: (value: string) => {
+          return value.length === 0 || formatTargetDir(value).length > 0
+            ? undefined
+            : "Invalid project name";
+        },
+      });
+      if (prompts.isCancel(projectName)) return cancel();
+      targetDir = formatTargetDir(projectName);
+    } else {
+      targetDir = defaultProjectName;
+    }
+  }
+
+  // 2. Handle directory if exist and not empty
+  if (fs.existsSync(targetDir) && !isEmpty(targetDir)) {
+    let overwrite: "yes" | "no" | undefined = argOverwrite ? "yes" : undefined;
+    if (!overwrite) {
+      if (interactive) {
+        const res = await prompts.select({
+          message:
+            (targetDir === "."
+              ? "Current directory"
+              : `Target directory "${targetDir}"`) +
+            ` is not empty. Please choose how to proceed:`,
+          options: [
+            {
+              label: "Cancel operation",
+              value: "no",
+            },
+            {
+              label: "Remove existing files and continue",
+              value: "yes",
+            },
+          ],
+        });
+        if (prompts.isCancel(res)) return cancel();
+        overwrite = res;
+      } else {
+        overwrite = "no";
+      }
+    }
+
+    switch (overwrite) {
+      case "yes":
+        emptyDir(targetDir);
+        break;
+      case "no":
+        cancel();
+        return;
+    }
+  }
+
+  const root = path.join(cwd, targetDir);
+
+  // 3. Clone the repository
+  prompts.log.step(`Cloning template from ${TEMPLATE_REPO}...`);
+
+  try {
```

Comment:
not sure the try catch here is necessary since run already exits with code = 1 in case of error, maybe run should throw ?

Reply:
Which has been inspired from https://github.com/vitejs/vite/blob/main/packages/create-vite/src/index.ts